Formula/python@3.9.rb

```diff
@@ -161,12 +163,32 @@ def install
       f.gsub! "DEFAULT_FRAMEWORK_FALLBACK = [", "DEFAULT_FRAMEWORK_FALLBACK = [ '#{HOMEBREW_PREFIX}/Frameworks',"
     end
 
+    ENV["CPATH"] = cpath.join(":") unless cpath.empty?
     args << "CFLAGS=#{cflags.join(" ")}" unless cflags.empty?
     args << "LDFLAGS=#{ldflags.join(" ")}" unless ldflags.empty?
     args << "CPPFLAGS=#{cppflags.join(" ")}" unless cppflags.empty?
 
     system "./configure", *args
     system "make"
+    system "bash", "-c", "DYLD_FRAMEWORK_PATH=#{Dir.pwd} ./python.exe -c 'import dbm; print(dbm)'"
+    File.open("dbm_test.py", "w") do |f|
+      f.write <<~EOS
+        print('68528 1')
+        import dbm
+        print('68528 2')
+        print(dbm)
+        print(dbm.ndbm)
+        print('68528 3')
+        with dbm.ndbm.open("test", "c") as db:
+            print('68528 4')
+            db[b"foo \\xbd"] = b"bar \\xbd"
+            print('68528 5')
+        print('68528 6')
+      EOS
+    end
+    system "bash", "-c", "DYLD_FRAMEWORK_PATH=#{Dir.pwd} ./python.exe dbm_test.py"
+    system "echo", "done"
+    system fail_here
```

Comment:
The diagnostics you're trying to do here are getting increasingly elaborate... Maybe you can explain what it is you want to find out here?

Reply:
Heh, yeah, I'm trying to diagnose why [this test](https://github.com/Homebrew/homebrew-core/blob/master/Formula/python@3.9.rb#L423) is failing on Arm after my changes.
The only logs it shows when it breaks are:
```
/usr/bin/curl --version
/usr/libexec/java_home --xml --failfast
```
So I need to go step-by-step to see which part is causing the error. Apparently it's able to import `dbm` and use `dbm.ndbm`, so I'll be able to see which specific line is imploding in another few CI runs.

Reply:
Check the uploaded CI artifacts upon failure. Those should contain more detailed build logs.

Reply:
@carlocab [here's an example run where the logs don't have enough information](https://github.com/Homebrew/homebrew-core/pull/68528/checks?check_run_id=1704222182).
Where can I see the uploaded CI artifacts?

Reply:
They haven't been uploaded because the run hasn't been completed. All the runners have to finish before anything is uploaded... I've cancelled the run now so it should upload any logs it has once the remaining runners are done cleaning up.

Reply:
Oh, I see how that works now, thanks!
Strangely, I'm not seeing a stack trace in `04.bash`:
```
2021-01-14 14:33:30 -0500


bash
-c
DYLD_FRAMEWORK_PATH=/private/tmp/python-3.9-20210114-70167-1qej46z/Python-3.9.1 ./python.exe dbm_test.py


HOMEBREW_VERSION: 2.7.4-17-ge989a0f
<snip rest of homebrew context dump>
```

FWIW, I know that I'm calling python correctly because it works in `03.bash`:
```
2021-01-14 14:33:30 -0500

bash
-c
DYLD_FRAMEWORK_PATH=/private/tmp/python-3.9-20210114-70167-1qej46z/Python-3.9.1 ./python.exe -c 'import dbm; print(dbm)'

<module 'dbm' from '/private/tmp/python-3.9-20210114-70167-1qej46z/Python-3.9.1/Lib/dbm/__init__.py'>
```